paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc

```diff
@@ -61,414 +61,428 @@ static mkldnn::memory::data_type GetDstType(bool is_int8, bool is_bfloat16,
 
-      auto paddings_temp = ctx.Attr<std::vector<int>>("paddings");
-      std::vector<int64_t> paddings(begin(paddings_temp), end(paddings_temp));
+    auto paddings_temp = ctx.Attr<std::vector<int>>("paddings");
+    std::vector<int64_t> paddings(begin(paddings_temp), end(paddings_temp));
 
-      auto dilations_temp = ctx.Attr<std::vector<int>>("dilations");
-      std::vector<int64_t> dilations(begin(dilations_temp),
-                                     end(dilations_temp));
+    auto dilations_temp = ctx.Attr<std::vector<int>>("dilations");
+    std::vector<int64_t> dilations(begin(dilations_temp), end(dilations_temp));
 
-      UpdatePaddingAndDilation(&paddings, &dilations, padding_algorithm,
-                               data_dims, strides, ksize);
+    UpdatePaddingAndDilation(&paddings, &dilations, padding_algorithm,
+                             data_dims, strides, ksize);
 
-      std::transform(dilations.begin(), dilations.end(), dilations.begin(),
-                     [](int64_t i) { return i - 1; });
+    std::transform(dilations.begin(), dilations.end(), dilations.begin(),
+                   [](int64_t i) { return i - 1; });
 
-      const auto src_tz = framework::vectorize(input->dims());
+    const auto src_tz = framework::vectorize(input->dims());
 
-      auto weights_tz = framework::vectorize(filter->dims());
-      platform::GetGroupConvWeightsTz(weights_tz, groups);
+    auto weights_tz = framework::vectorize(filter->dims());
+    platform::GetGroupConvWeightsTz(weights_tz, groups);
 
-      const auto dst_tz = framework::vectorize(output->dims());
+    const auto dst_tz = framework::vectorize(output->dims());
+
+    const mkldnn::memory::dims stride_dims = strides;
+    const auto mkldnn_paddings = platform::ToMkldnnPadding(paddings);
+    const mkldnn::memory::dims dilations_dims = dilations;
+
+    /* create memory descriptor for convolution without specified format
+     * ('any') which lets a primitive (convolution in this case) choose
+     * the memory format preferred for best performance
+     */
+    auto chosen_memory_format = MKLDNNMemoryFormat::any;
+    auto data_type = mkldnn::memory::data_type::f32;
+    if (ctx.Attr<std::string>("mkldnn_data_type") == "bfloat16" ||
+        std::is_same<T_out, platform::bfloat16>::value)
+      data_type = mkldnn::memory::data_type::bf16;
+
+    mkldnn::memory::desc src_md, weights_md;
+    if (platform::is_int8<T>()) {
+      src_md = platform::MKLDNNMemDesc(
+          src_tz, framework::ToMKLDNNDataType(input->type()),
+          chosen_memory_format);
+      weights_md = platform::MKLDNNMemDesc(
+          weights_tz, mkldnn::memory::data_type::s8, chosen_memory_format);
+    } else {
+      src_md = platform::MKLDNNMemDesc(src_tz, data_type, chosen_memory_format);
+      weights_md = platform::MKLDNNMemDesc(weights_tz, data_type,
+                                           MKLDNNMemoryFormat::any);
```

Comment:
You are using "chosen memory format" everywhere except that place, please unity that

Reply:
good catch. Fixed